### ISSUE TITLE
Dotfiles still being ignored with version 6.8.0

### ISSUE THREAD

**Issue by aryzing** [eslint/eslint#12938]
**Tell us about your environment**

* **Node version**: v13.7.0
* **npm version**: v6.13.6
* **Local ESLint version**: v6.8.0 (Currently used)
* **Global ESLint version**: Not found

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
const defaults = {
    "env": {
        "browser": true,
        "es6": true
    },
    "extends": "eslint:recommended",
    "globals": {
        "Atomics": "readonly",
        "SharedArrayBuffer": "readonly"
    },
    "parserOptions": {
        "ecmaVersion": 2018,
        "sourceType": "module"
    },
    "rules": {
    },
}

module.exports = {
    ...defaults,
    overrides: [{
        files: ['.my-dotfile.js'],
        ...defaults
    }]
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

Dot files are being ignored, even when explicitly set in the `overrides` section. See config.

```bash
yarn run eslint .my-dotfile.js # ignores file despite explicitly naming it
yarn run eslint .              # ingores file too despite being explicitly set in overrides section in config
```

**What did you expect to happen?**

The `.my-dotfile.js` file should have been linted

**What actually happened? Please include the actual, raw output from ESLint.**

It did not lint the file

```
/home/ayzing/workspace/repro-eslint-dotfiles-not-working/.my-dotfile.js
  0:0  warning  File ignored by default.  Use a negated ignore pattern (like "--ignore-pattern '!<relative/path/to/filename>'") to override

✖ 1 problem (0 errors, 1 warning)
```

**Are you willing to submit a pull request to fix this bug?**

Not at this time. Here's a [sample repo](https://github.com/aryzing/repro-eslint-dotfiles-not-working) to help with the debugging.

**Comment by kaicataldo** [comment]
`overrides` still respects ignore patterns, so you'll want to follow the suggestion given in the error message. Alternatively, you can use an [`.eslintignore` file](https://eslint.org/docs/user-guide/configuring#eslintignore) or by specifying [`ignorePatterns`](https://eslint.org/docs/user-guide/configuring#ignorepatterns-in-config-files) in your configuration.

Unrelated to this, but overrides are merged into the top-level config (the intention is to allow users to override the base config for specific files), so you don't need to merge `defaults` into the `overrides` entry.

**Comment by thescientist13** [comment]
For what it's worth @aryzing , I had been looking to exactly this, but to lint my _.eslintrc.js_ file and wasn't getting it work.  I saw @kaicataldo 's reply and did what he suggested and added _.eslintrc.js_ to my _.eslintignore_ file and ESLint picked it up.

#### .eslintrc.js
```javascript
module.exports = {
  /* */
}
```

#### .eslintignore
```shell
**/node_modules/**
!.eslintrc.js
```

**Comment by kaicataldo** [comment]
@thescientist13 Glad you got it to work! That's something that's bitten us all at one time or another. Starting in [v7.0.0](https://eslint.org/blog/2020/02/whats-coming-in-eslint-7.0.0#updated-default-ignore-patterns), `.eslintrc.js` won't be ignored by default!

**Comment by thescientist13** [comment]
Great, and good to know @kaicataldo !  And now also answers a very important question... _who lints the linters?_ . (spoilers: it's also the linters 😄 )

**Comment by aryzing** [comment]
Thanks @kaicataldo and @thescientist13 for the valuable info. My confusion was coming from reading [Overrides in a config file can now match dotfiles](https://eslint.org/docs/user-guide/migrating-to-6.0.0#overrides-in-a-config-file-can-now-match-dotfiles) and [this post](https://github.com/eslint/eslint/issues/4828#issuecomment-194535273)
